tsconfig.json

```diff
@@ -1,61 +1,64 @@
 {
```

Comment:
I now recommend to base the tsconfig on a predefined one like I do in https://github.com/zwave-js/node-zwave-js/blob/3a7251d111320190f8936b277a723c0b0ae9711f/tsconfig.json#L2

This one requires `@tsconfig/node12` to be installed as a `devDepencency`, and makes targeting a different node version easier. For the options:

target/module/lib is not necessary if you extend a shared config
allowJs/checkJs is okay for the mixed codebase

You should add this to power project references (TS monorepo), and probably remove it from the subproject tsconfig:
```
		"composite": true,
		"declaration": true,
		"declarationMap": true,
```

I'm not 100% sure about `noEmit`.

You'll also want to add
```
"importsNotUsedAsValues": "error",
```
to avoid a runtime effect when you only wanted to import a type.

Debugging the TS part becomes easier if you use
```
		"sourceMap": true,
		"inlineSourceMap": false,
```

This
```
		"stripInternal": true,
```
is nice if you need public properties for use from elsewhere within the same package but don't want to leak it in the type declarations. You can then do this:
```ts
class Foo {
  /** @internal */
  public function doSthInternal() { ... }
}

// same package:
const foo = new Foo();
foo.doSthInternal(); // no error

// other package:
const foo = new Foo();
foo.doSthInternal(); // error!
```

This
```
"skipLibCheck": true
``` 
is almost always required to avoid erroring on incorrect type declarations in dependencies.

Not sure about `resolveJsonModule`, do we require JSONs anywhere?

```
"noImplicitAny": false
```
is not a good idea unless you just need to make something work **now**.

Reply:
I hope I got everything right, checkJS true now produced several hunderts of errors in adapter.js so I deactivated it. Maybe you can have another look if you'll find some time.

Reply:
> checkJS true now produced several hunderts of errors in adapter.js

I wouldn't have expected anything else. Maybe you now see why my attempts to introduce typings in one go never went very far 🙈
We can enable it from time to time and chew threw the errros slowly.